tests/test_contentindexparser.py

```diff
@@ -636,16 +677,75 @@ def test_example1_csv(self):
         ci_parser = ContentIndexParser(sheet_reader, "tests.input.example1.nestedmodel")
         self.check_example1(ci_parser)
 
+    def test_example1_csv_composite(self):
+        # Same test as test_generate_flows but with csvs
+        sheet_reader = CSVSheetReader(self.input_dir / "content_index.csv")
+        reader = CompositeSheetReader([sheet_reader])
+        ci_parser = ContentIndexParser(reader, "tests.input.example1.nestedmodel")
+        self.check_example1(ci_parser)
+
     def test_example1_split_csv(self):
         # Same test as test_generate_flows but with csvs
-        sheet_reader = CSVSheetReader(self.input_dir / "content_index1.csv")
-        ci_parser = ContentIndexParser(sheet_reader, "tests.input.example1.nestedmodel")
-        sheet_reader = CSVSheetReader(self.input_dir / "content_index2.csv")
-        ci_parser.add_content_index(sheet_reader)
+        sheet_reader1 = CSVSheetReader(self.input_dir / "content_index1.csv")
+        sheet_reader2 = CSVSheetReader(self.input_dir / "content_index2.csv")
+        reader = CompositeSheetReader([sheet_reader1, sheet_reader2])
+        ci_parser = ContentIndexParser(reader, "tests.input.example1.nestedmodel")
         self.check_example1(ci_parser)
 
     def test_example1_xlsx(self):
         # Same test as above
         sheet_reader = XLSXSheetReader(self.input_dir / "content_index.xlsx")
         ci_parser = ContentIndexParser(sheet_reader, "tests.input.example1.nestedmodel")
         self.check_example1(ci_parser)
+
+    def test_example1_xlsx_composite(self):
+        # Same test as above
+        sheet_reader = XLSXSheetReader(self.input_dir / "content_index.xlsx")
+        reader = CompositeSheetReader([sheet_reader])
+        ci_parser = ContentIndexParser(reader, "tests.input.example1.nestedmodel")
+        self.check_example1(ci_parser)
+
+
+class TestMultiFile(TestTemplate):
```

Comment:
What is being tested here? If it's the composite reader, then I think we can do it with mock readers and not introduce more binary (XLSX) files whose contents we cannot track with git.

Reply:
I want to make sure that the CompositeReader works with XLSX readers specifically, as these behave different from CSVs (and similarly to Google Sheets).

Reply:
I now created another abstract class for readers that have a dictionary of their sheets (GoogleSheet, XLSX, Mock). As Mock and XLSX now have the same parent that implements the methods of interest, so I now have more confidence that using the Mock is sufficient for testing this.